pkg/controller/syncconfig/utils.go

```diff
@@ -0,0 +1,113 @@
+package syncconfig
+
+import (
+	"context"
+	"strings"
+
+	templateapi "github.com/openshift/origin/pkg/template/apis/template"
+	"github.com/openshift/origin/pkg/template/generator"
+	"github.com/openshift/origin/pkg/template/templateprocessing"
+	"github.com/vshn/espejo/pkg/apis/sync/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func recreateOject(ctx context.Context, unstructObj *unstructured.Unstructured, c client.Client) error {
+	unstructObj.SetResourceVersion("")
+
+	err := c.Delete(ctx, unstructObj, client.DeleteOptionFunc(func(d *client.DeleteOptions) {
+		option := metav1.DeletePropagationForeground
+		d.PropagationPolicy = &option
+	}))
+
+	if err != nil {
+		return err
+	}
+
+	err = c.Create(context.Background(), unstructObj)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func processTemplate(unstructObj *unstructured.Unstructured, projectName string) error {
+	params := make(map[string]templateapi.Parameter, 1)
+	params["PROJECT_NAME"] = templateapi.Parameter{
+		Name:  "PROJECT_NAME",
+		Value: projectName,
+	}
+
+	generatorMap := make(map[string]generator.Generator)
+	processor := templateprocessing.NewProcessor(generatorMap)
+
+	_, err := processor.SubstituteParameters(params, unstructObj)
+
+	return err
+}
+
+func getFullName(unstructuredObject *unstructured.Unstructured) []interface{} {
```

Comment:
The function name implies a return value of type string. Something along `getLoggingKeysAndValues()` would be more explicit about the intent.